### PR TITLE
[MULTI] Fixed devices comparison to respect default device id

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - [GPU] Changed default value for DEVICE_ID key from "" to "0" to allow passing default value back to plugin in SetConfig call
 - [MULTI] Fixed device name comparison logic in multi plugin to process device ID correctly.

### Tickets:
 - *62004*

### CODE REVIEW

File: inference-engine/src/multi_device/multi_device_async_infer_request.cpp [hunk 1]
```diff
@@ -47,7 +48,9 @@ MultiDeviceAsyncInferRequest::MultiDeviceAsyncInferRequest(
                                _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cbegin(),
                                _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cend(),
                                [&name](const MultiDevicePlugin::DeviceInformation& d){
-                                    return d.deviceName == name; });
+                                    DeviceIDParser deviceParser(name);
+                                    return d.deviceName == name || d.deviceName == deviceParser.getDeviceName();
```
Comment (ilya-lavrenov) [inline inference-engine/src/multi_device/multi_device_async_infer_request.cpp:52]: probably, we should compare `r->getDeviceName()` with `deviceName + execNetwork.GetConfig(CONFIG_KEY(DEVICE_ID)).as<string>()`

In this case we can match exactly devices

File: inference-engine/src/multi_device/multi_device_async_infer_request.cpp [hunk 2]
```diff
@@ -35,31 +38,42 @@ MultiDeviceAsyncInferRequest::MultiDeviceAsyncInferRequest(
...
+                                [&name](const MultiDevicePlugin::DeviceInformation& d) {
+                                    if (d.deviceName == name)
+                                        return true;
 
-                       } else {
+                                    // In case if remote blob always returns fully-specified device name (xPU.ID)
+                                    // we try to append default device ID to the name in DeviceInformation and compare again
+                                    DeviceIDParser parsed{d.deviceName};
+                                    if (parsed.getDeviceID().empty() && !d.defaultDeviceID.empty()) {
+                                        return (d.deviceName + "." + d.defaultDeviceID) == name;
```
Comment (ilya-lavrenov) [inline inference-engine/src/multi_device/multi_device_async_infer_request.cpp:60]: @myshevts should not we use such compare in MultiExecNet.SetConfig?

Comment (ilya-lavrenov) [inline inference-engine/src/multi_device/multi_device_async_infer_request.cpp:60]: @myshevts should not we use such comparison in MultiExecNet.SetConfig?

File: inference-engine/src/multi_device/multi_device_async_infer_request.cpp (hunk 1 above)
Comment (vladimir-paramuzov) [reply inference-engine/src/multi_device/multi_device_async_infer_request.cpp]: Fixed

File: inference-engine/src/multi_device/multi_device_async_infer_request.cpp (hunk 2 above)
Comment (myshevts) [reply inference-engine/src/multi_device/multi_device_async_infer_request.cpp:60]: we discussed with Vladimir that this code should be move from scheduling to the ParseMetaDevices

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/multi_device/multi_device_exec_network.cpp [hunk 3]
```diff
@@ -372,6 +372,10 @@ std::shared_ptr<InferenceEngine::RemoteContext> MultiDeviceExecutableNetwork::Ge
                              << " Current list of devices allowed via the DEVICE_PRIORITIES config: " << devices_names;
 }
 
+std::shared_ptr<InferenceEngine::ICore> MultiDeviceExecutableNetwork::GetCore() const {
```
Comment (myshevts) [inline inference-engine/src/multi_device/multi_device_exec_network.cpp:375]: minor but this method is not needed anymore

**Review by myshevts — APPROVED** [review]

**Pull request merged**